lib/pure/asyncfutures.nim

```diff
@@ -30,7 +33,14 @@ type
 when not defined(release):
   var currentID = 0
 
-proc callSoon*(cbproc: proc ()) {.gcsafe.}
+var callSoonProc* {.threadvar.}: (proc(cbproc: proc ()) {.gcsafe.})
+
+proc callSoon*(cbproc: proc ()) =
+  if callSoonProc == nil:
+    # Loop not initialized yet. Call the function directly to allow setup code to use futures.
```

Comment:
(Note that GlobalDispatcher is a misleading name, it actually should be called ThreadDefaultDispatcher)